### ISSUE TITLE
rg.pyexecute: REQUIREMENTS needs a better usage example

### ISSUE THREAD

**Issue by gkorland** [RedisGears/RedisGears#282]
```
RG.PYEXECUTE "<function>" [REQUIREMENTS "<dep> ..."] [UNBLOCKING]
```
https://oss.redislabs.com/redisgears/commands.html#rgpyexecute

**Comment by nareshn842** [comment]
Hi, Is there any update on this . Can you please let me know how to pass the requirements to Gear function.

**Comment by MeirShpilraien** [comment]
As the docs says: RG.PYEXECUTE <your python code> REQUIREMENTS <req1> <req2> ...

**Comment by nareshn842** [comment]
Hi, Thank you for the reply.

I am trying to execute the command from python file (client) not from the redis-cli.  What ever the command above shared is when we try to execute from cli right.? Please correct my understanding.

Here is the snippet what I am trying .

con = redis.StrictRedis(host=redis_host, port=redis_port, password=redis_password)
script = open('process_redis_stream.py', 'r').read()
con.execute_command("RG.PYEXECUTE", script, "REQUIREMENTS=redis")


Gear function register logic  is in "process_redis_stream.py" which I am reading it from another python file. At this moment I need to to pass the requirements.

but execute_command is only taking two arguments. Invoking from a python client details not given in documentation .Can you please throw some light on this.

Thank You
Naresh.

**Comment by MeirShpilraien** [comment]
execute_command can take as many arguments as you want. Basically, it needs to look like this:
`con.execute_command("RG.PYEXECUTE", script, "REQUIREMENTS", "redis")`
Side note: I am not sure why you need Redis client on the server-side, can you share some more details about it?

**Comment by nareshn842** [comment]
Hi,

We need to process the stream data and store the result in another data base thats why need a client. 

Now I am able to pass the requirements but throwing some error due to proxy.  Is there any way how to pass proxy details to execute_command..?

Docker Error logs.
---------------
1:M 03 Mar 2021 08:47:21.418 # <module> Execution failed command : /bin/bash -c "cd '/var/opt/redislabs/modules/rg//python3_1.0.6//redis';/var/opt/redislabs/modules/rg//python3_1.0.6//bin/python3 -m pip wheel 'redis'"

1:M 03 Mar 2021 08:47:22.419 * <module> Executing : /bin/bash -c "cd '/var/opt/redislabs/modules/rg//python3_1.0.6//redis';/var/opt/redislabs/modules/rg//python3_1.0.6//bin/python3 -m pip wheel 'redis'"

WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f9e04e88c50>, 'Connection to pypi.org timed out. (connect timeout=15)')': /simple/redis/

WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'ConnectTimeoutError(<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7f9e04e88ef0>, 'Connection to pypi.org timed out. (connect timeout=15)')': /simple/redis


Thank You.
Naresh

**Comment by MeirShpilraien** [comment]
There is no such way but you can install the requirement in outside environment without proxy and then export and import this requirement using gears-cli, take a look here for more information:
https://forum.redislabs.com/t/how-to-install-python-packages-to-redisgears-docker-image/825/4

**Comment by nareshn842** [comment]
Hello,

Thanks for the reply . We have bypass issue using gears-cli.  

Apart from passing requirements is there any way to pass some extra arguments and get those arguments in gear function.

Myarg1="Hello"
Myarg2="World"

con.execute_command("RG.PYEXECUTE", script, "REQUIREMENTS", "redis",Myarg1,Myarg2)

when the script executed gear function should receive these arguments.

Thank You,
Naresh

**Comment by MeirShpilraien** [comment]
Yes but the way to do this is register a gear function using the command reader and then trigger it with commands and arguments. Example:
```
127.0.0.1:6379> RG.PYEXECUTE "GB('CommandReader').map(lambda x: len(x)).register(trigger='CountArgs')"
OK
127.0.0.1:6379> RG.TRIGGER CountArgs 1 2 3 4
1) "5"
```
The record that passes to the execution triggered by RG.TRIGGER command is a list of all the arguments given in the command (including the trigger name) in our example: `['CountArgs', '1', '2', '3', '4']`
To read more about the command reader: https://oss.redislabs.com/redisgears/readers.html#commandreader

**Comment by nareshn842** [comment]
Hi,

But we have requirement to read stream reader and gear function listen to the specified stream.

how we can communicate from one gear to another gear..? values passed form command reader to stream reader.?


Thanks,
Naresh

**Comment by MeirShpilraien** [comment]
Basically, you can send a gear that registers both the command reader and the stream reader.. Can you be more specific about the usecase? What you need from the command reader and what you need from the stream reader?

**Comment by nareshn842** [comment]
We need stream reader only please find use case details below.

Use case:  Gear function running in one data base  read the the stream data after processing the stream data store the results  in  another database.  

My gear registration logic in one python file (script.pty) which is invoked from another  python client.

con.execute_command("RG.PYEXECUTE", script, "REQUIREMENTS", "redis)

This is how we did it . but as per use case  to store the result to another data base we cant hardcode the DB details inside gear . We would like to pass the DB details dynamically to gear function.

Thanks,
Naresh

**Comment by MeirShpilraien** [comment]
I see, so for such things you can use the `configGet` api (https://oss.redislabs.com/redisgears/runtime.html#configget). You can give the db info as configuration values to RedisGears (when loading the module) and retrieve them inside your script
